automation_tools/__init__.py

```diff
@@ -2396,10 +2354,14 @@ def fix_hostname(entry_domain=None, host_ip=None):
             .format(host_ip, entry_domain, host))
     else:
         # Required for fixing the hostname when using satellite-installer
-        ip_addr = run(r"ping -c1 $(hostname) | awk -F\( '/icmp_seq/{print$2}' "
-                      r"| awk -F\) '{print$1}'")
-        run('echo "{0} $(hostname) $(hostname -s)" >> /etc/hosts'
-            .format(ip_addr))
+        for ip_ver in (4, 6):
```

Comment:
`fix_hostname()`:
- now handles also IPv6 fix if IPv6 address exists on the host
- does the fix to `/etc/hosts` only if IP is not reversibly resolvable to hostname (so no hacky fix when we have proper DNS records created)